chain/client/src/view_client.rs

```diff
@@ -172,12 +204,130 @@ impl ViewClientActor {
         finality: &Finality,
     ) -> Result<CryptoHash, near_chain::Error> {
         match finality {
-            Finality::None => Ok(self.chain.head()?.last_block_hash),
-            Finality::DoomSlug => Ok(*self.chain.head_header()?.last_ds_final_block()),
-            Finality::Final => Ok(self.chain.final_head()?.last_block_hash),
+            Finality::None => Ok(self.hot_chain.head()?.last_block_hash),
+            Finality::DoomSlug => Ok(*self.hot_chain.head_header()?.last_ds_final_block()),
+            Finality::Final => Ok(self.hot_chain.final_head()?.last_block_hash),
         }
     }
 
+    /// Determines temperature of a block with given block hash.
+    ///
+    /// Returned temperature allows fetching all data related to block at given
+    /// height as well as data about previous and next blocks (if any).
+    fn get_temperature(&self, _block_hash: &CryptoHash) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of a block with given block height.
+    ///
+    /// Returned temperature allows fetching all data related to block with
+    /// given hash as well as data about previous and next blocks (if any).
+    fn get_temperature_by_height(
+        &self,
+        _block_height: BlockHeight,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of a chunk with given chunk hash.
+    fn get_temperature_by_chunk_hash(
+        &self,
+        _chunk_hash: &ChunkHash,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
+    }
+
+    /// Determines temperature of data from transaction hash.
+    fn get_temperature_by_tx_hash(
+        &self,
+        _tx_hash: &CryptoHash,
+        _sender_id: &AccountId,
+    ) -> Result<Temperature, near_chain::Error> {
+        // TODO(#6119): Implement temperature detection.
+        Ok(Temperature::Hot)
```

Comment:
Could you briefly explain how these get_temperature functions will be implemented? Is there any kind of guarantees like the most recent five epochs are hot and anything older is cold? Or does it depend on the gc process?

Reply:
Hot storage should be like rpc storage with maybe a bit slower gc. GC for hot storage will not go past the head of cold storage, but also will not go past the specified number of epochs. 
It is still TBD, but I think we will copy data to cold storage in epochs. Canonical chain cleaning on some epoch will start once that whole epoch is in cold storage.
This way we can always determine which chain is the right one in these functions by looking at the head of cold storage. 
There is also a complication with GC and ViewClient, it is explained in description here https://github.com/near/nearcore/issues/8001

Reply:
I see. Is it true that like GC, data is copied to cold storage by block? If I want to check the temperature for a chunk or receipt, I first need to determine the block hash of the chunk or receipt corresponds to, then get the temperature of that. If that's the case, I find all these get_tempature_by_XXX functions a bit redundant and confusing, maybe what you need here is get_block_hash_by_XXX and computer temperature from the block hash.

Reply:
It is also still unclear to me how to calculate for example, block hash from receipt hash without knowing the temperature. Do we fist try to look in hot storage, and if it doesn't exist, check in cold storage?

Reply:
> Do we fist try to look in hot storage, and if it doesn't exist, check in cold storage? 
I think right now we don't have better ideas

Reply:
The logic to determine temperature needs to be somewhere.  The idea behind get_temperature methods is to collect all that logic in a few well-defined places.